aips/aip-4.md

```diff
@@ -0,0 +1,78 @@
+
+### Rationale
+
+Extending this utility to the existing Aptos-Token standard provides significant benefits:
+
+- Consensus on a soulbound token standard
+- Monitoring a single event store for soulbound and non-soulbound tokens
+- Compatibility with the Aptos-Token standard.
+
+An alternative solution to the non-transferability problem is creating a "Locking Standard", where tokens can be deposited, claimed, and burned. Creating a locker that acts as a wrapper around tokens essentially makes them soulbound. This solution has a significant drawback: monitoring can no longer be handled for both soulbound and non-soulbound tokens in a single event store, thus creating the need for separate monitoring of an entirely different event store. This is undesirable from usability, composability, and scalability standpoints.
+
+### Specification
+
+Modifications required to the current [Aptos-Token](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token/sources/token.move) implementation:
+
+- Append “soulbound: bool” too *Tokens*
+- Minting of soulbound tokens can only be handled through *“claims”*. After a soulbound NFT has been claimed it is locked to the receiver with the only valid operation being *“burn”,* if applicable to the token’s configuration*.*
+- Validate before any transfer execution to ensure the token isn’t soulbound.
+
+```rust
+/// Executed when a specific token_id is set or updated to soulbound, according to the parameter "bounded".
+public fun soulbound(token_id: TokenId, bounded: bool)
+
+/// Returns true if soulbound, false if non-soulbound.
+public fun get_soulbound(token_id: TokenId)
+```
+
+### Future Applications
+
+A common use case for Soulbound NFTs is Proof of Attendance, where soulbound NFTs are distributed upon attendance of an event. Often event coordinators have all attendees’ addresses at the time of distribution, in which case iteratively calling “*offer”* to allow attendees to claim their token costs unnecessary gas and network usage. **A simple solution to this use case is implementing a function to “offer_batch” that may look as follows:
+
+```rust
+public fun offer_batch(offers: vector<{
+	sender: signer,
+	receiver: address,
+  creator: address,
+  collection: String,
+  name: String,
+  property_version: u64,
+  amount: u64,
+}>) acquires PendingClaims {
+
+	while (i < vector::length(&offers)){
+		offer_script(&sender, receiver, creator, collection, name, property_version, amount);
+	}
+
+}
+```
+
+**** This function allows multiple offers to be sent in a single transaction.
+
+### Suggested I**mplementation Timeline**
+
+To be determined.
```

Comment:
I think this is one of those things where you should try a couple implementations and come back with the pros and cons. It isn't clear to me that there is a perfect solution here.

Reply:
Will do, working through it with @areshand :)